{"owner":{"reputation":16058,"user_id":240512,"user_type":"registered","profile_image":"https://i.stack.imgur.com/Ew6lG.png?s=128&g=1","display_name":"MSDN.WhiteKnight","link":"https://ru.stackoverflow.com/users/240512/msdn-whiteknight"},"is_accepted":true,"score":1,"last_activity_date":1551803210,"creation_date":1551803210,"answer_id":952776,"question_id":952304,"body":"<p>Сообщение анализатора кода <a href=\"https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2240-implement-iserializable-correctly?view=vs-2017\" rel=\"nofollow noreferrer\">CA2240</a> в данном случае выводится из-за того, что метод GetObjectData не помечен как переопределяемый (виртуальный). Так как базовый класс Exception содержит свою реализацию GetObjectData, необходимо также добавить override. Правильное объявление: </p>\n\n<pre><code>void virtual GetObjectData(SerializationInfo^ info, StreamingContext^ context) override\n</code></pre>\n"}